src/tls.c

```diff
@@ -7306,12 +7306,16 @@ static int TLSX_KeyShare_ProcessOqs(WOLFSSL* ssl, KeyShareEntry* keyShareEntry)
 
     if (ssl->options.side == WOLFSSL_SERVER_END) {
         /* I am the server, the shared secret has already been generated and
-         * is in keyShareEntry->ke; transfer ownership of the buffer. */
-        if (ssl->arrays->preMasterSecret != NULL)
-            XFREE(ssl->arrays->preMasterSecret, ssl->heap,
-                  DYNAMIC_TYPE_SECRET);
-        ssl->arrays->preMasterSecret = keyShareEntry->ke;
+         * is in keyShareEntry->ke; copy it to the pre-master secret
+         * pre-allocated buffer. */
+        if (keyShareEntry->keLen > ENCRYPT_LEN) {
+            WOLFSSL_MSG("shared secret is too long.\n");
```

Comment:
Old code free'd `XFREE(ssl->arrays->preMasterSecret, ssl->heap, DYNAMIC_TYPE_SECRET);`.  Also the WOLFSSL_MSG already adds `\n`.

Reply:
fixed here and below.